### PR TITLE
Metal hillshade layer

### PR DESCRIPTION
**Pull request by alexcristici**
Hillshade layer on Metal.

### CODE REVIEW

**Review by mwilsnd — APPROVED** [review]

File: include/mbgl/shaders/mtl/hillshade.hpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    /*if (permutation.overdrawInspector) {
+        return half4(1.0);
+    }*/
+
+    float4 pixel = image.sample(image_sampler, in.pos);
+
+    float2 deriv = ((pixel.rg * 2.0) - 1.0);
+
+    // We divide the slope by a scale factor based on the cosin of the pixel's approximate latitude
+    // to account for mercator projection distortion. see #4807 for details
```
Comment (louwers) [inline include/mbgl/shaders/mtl/hillshade.hpp:81]: I know this comment is also in the GL version, but I am not sure what it refers to.

File: include/mbgl/shaders/mtl/hillshade.hpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    };
+}
+
+half4 fragment fragmentMain(FragmentStage in [[stage_in]],
+                            device const HillshadeDrawableUBO& drawable [[buffer(2)]],
+                            device const HillshadeEvaluatedPropsUBO& props [[buffer(3)]],
+                            texture2d<float, access::sample> image [[texture(0)]],
+                            sampler image_sampler [[sampler(0)]]) {
+
+    /*if (permutation.overdrawInspector) {
```
Comment (louwers) [inline include/mbgl/shaders/mtl/hillshade.hpp:72]: Is this purposely left in?

**Review by louwers — APPROVED** [review]

**Review by sjg-wdw — APPROVED** [review]

File: include/mbgl/shaders/mtl/hillshade.hpp (hunk 2 above)
Comment (alexcristici) [reply include/mbgl/shaders/mtl/hillshade.hpp:72]: Good catch, I need to fix this. 👍

Comment (alexcristici) [reply include/mbgl/shaders/mtl/hillshade.hpp]: Fixed.

File: include/mbgl/shaders/mtl/hillshade.hpp (hunk 1 above)
Comment (alexcristici) [reply include/mbgl/shaders/mtl/hillshade.hpp:84]: I didn't focus to understand the shaders, just converted them from GL to Metal.

**Pull request merged**